### ISSUE TITLE
FIXED IN 3.0.19-unstable.6 - Auth.completeNewPassword seems to leave Auth in unstable state

### ISSUE THREAD

**Issue by JohnAtFenestra** [aws-amplify/amplify-js#6158]
Adding commentary from my own reply. This bug no longer evidences in 3.0.19-unstable.6 from https://cdnjs.com/libraries/aws-amplify

Original __completeNewPassword followed by signOut followed by immediate signIn produces a NotAuthorizedException__

Renaming to describe the scenario more accurately and to hopefully get a response

Yesterday, I opened this as a discussion, but I carried out an experiment and now am convinced I have found a bug.

**Describe the bug**
While conducting a simple end-to-end test with Amplify in Jest, I discovered an inexplicable bug scenario. All of this is demonstrated in the below example Javascript app.

**To Reproduce**

1. Create a new user
2. Auth.signIn using the temporaryPassword
3. Respond to the NEW_PASSWORD_REQUIRED challenge with Auth.completeNewPassword using a new password.
4. Auth.signOut
5. Auth.signIn using the new password - a NotAuthorizedException is returned

If one then runs an independent script which simply signs into the account with the new password, the signIn is successful. Also logs in successfully using HostedUI.

**Expected behavior**

The step 5 Auth.signIn should sign the user in successfully.

**Code Snippet**
```javascript

  it('should create a new user debuga', async () => {
    try {
      const userExists = await cognitoService.doesUserExist(newUser.username)
      if (userExists) {
        await cognitoService.deleteUser(newUser.username)
      }
      const createdUser = await cognitoService.createUser(newUser)
      expect(createdUser.User.Username).toEqual(newUser.username)
      expect(createdUser.User.UserStatus).toEqual('FORCE_CHANGE_PASSWORD')

      let user
      user = await Auth.signIn(newUser.username, newUser.temporaryPassword)
      expect(user.challengeName).toEqual('NEW_PASSWORD_REQUIRED')
      console.log('starting complete')
      if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
        user = await Auth.completeNewPassword(user, newUser.newPassword, {})
      }
      console.log('complete done')
      console.log('before signOut', await Auth.currentSession())
      await Auth.signOut()

      user = null

      console.log('Re Signin')
      user = await Auth.signIn(newUser.username, newUser.newPassword) // BLOWS UP HERE!!!
      console.log(user)

      const session = await Auth.currentSession()
      console.log(session.getIdToken().getJwtToken())
      // await cognitoService.deleteUser(newUser.username)
    } catch (error) {
      console.log('ERROR')
      console.log(error)
      console.log(error.stack)
      fail(error)
    }
  })

```

**Console output**

[output-with-debug.txt](https://github.com/aws-amplify/amplify-js/files/4826878/output-with-debug.txt)

**What is Configured?**
N/A
* Which steps did you follow via Amplify CLI when configuring your resources.
Did not use.

* Which resources do you have configured?

```
    cognito: {
      userPoolId: 'us-east-2_yLhk13q1L',
      region: 'us-east-2',
      userPoolWebClientId: '5e5g6mvl9a64o7v6gmho47vg2m',
    },
```

```javascript
      Amplify.configure({
        Auth: {
          region: cognitoConf.cognito.region,
          userPoolId: cognitoConf.cognito.userPoolId,
          userPoolWebClientId: cognitoConf.cognito.userPoolWebClientId,
          authenticationFlowType: 'USER_PASSWORD_AUTH',
        },
      })
 ```
* If applicable, provide more configuration data, for example for Amazon Cognito, run `aws cognito-idp describe-user-pool --user-pool-id us-west-2_xxxxxx` (Be sure to remove any sensitive data)

[user-pool.txt](https://github.com/aws-amplify/amplify-js/files/4826850/user-pool.txt) 

<details>
  <summary><strong>Environment</strong></summary>

```
 System:
    OS: Linux 5.4 Ubuntu 20.04 LTS (Focal Fossa)
    CPU: (8) x64 AMD Ryzen Embedded V1605B with Radeon Vega Gfx
    Memory: 1.80 GB / 30.88 GB
    Container: Yes
    Shell: 5.0.16 - /bin/bash
  Binaries:
    Node: 12.16.3 - ~/.nvm/versions/node/v12.16.3/bin/node
    Yarn: 1.22.4 - ~/.npm-packages/bin/yarn
    npm: 6.14.5 - ~/.nvm/versions/node/v12.16.3/bin/npm
  Browsers:
    Chrome: 83.0.4103.97
    Firefox: 77.0.1
  npmPackages:
    @babel/core: ^7.10.3 => 7.10.3 
    @babel/preset-env: ^7.10.3 => 7.10.3 
    @babel/preset-typescript: ^7.10.1 => 7.10.1 
    aws-amplify: ^3.0.18 => 3.0.18 
    aws-sdk: ^2.702.0 => 2.702.0 
    babel-jest: ^26.0.1 => 26.0.1 
    jest: ^26.0.1 => 26.0.1 
    node-fetch: ^2.6.0 => 2.6.0 
  npmGlobalPackages:
    @vue/cli: 4.3.1
    jest-cli: 26.0.1
    npm: 6.14.5

```

</details>

**Comment by JohnAtFenestra** [comment]
Adding this as documentation. I attempted to replicate this in a browser and stumbled across this CDN:

https://cdnjs.com/libraries/aws-amplify

Which supplies 3.0.19-unstable.6

This version did not generate the error I describe. I then replaced the src with 3.0.18 and the error showed up. The current node version is 3.0.18.

Is there a release schedule for the next version?

**Comment by amhinson** [comment]
@JohnAtFenestra Are you still seeing this error with the latest version released (3.0.21)?
